### ISSUE TITLE
Enhancements on Forms Layout

### ISSUE THREAD

**Issue by whizkidwwe1217** [vmware/clarity#1082]
**Select one ...**  (check one with "x")
```
[ ] bug
[ ] feature request
[x] enhancement
```

### Expected behavior

We're currently developing a web app for HRIS and we have tons of UI that extensively use forms to gather employee information. We are using Clarity-UI because of its simplicity, clarity and awesome components (Thank you so much for that! <3 ). But the only thing that this library don't have is the ability to layout form controls very nicely. I've attached a screen shot of the mockup design that I've made for form layout. 

![image](https://user-images.githubusercontent.com/1977391/27267431-d8013ff6-54da-11e7-8dfb-cd5efd8212ff.png)

### Actual behavior
There's too much padding between labels and controls:

![image](https://user-images.githubusercontent.com/1977391/27267625-61730700-54dc-11e7-96d0-8ff754734bcd.png)

![image](https://user-images.githubusercontent.com/1977391/27267745-46b3d7e0-54dd-11e7-8679-55fe9be1cb44.png)

### Environment details

* **Angular version:** 2.0.X

* **Clarity version:** 

* **OS and version:** 

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

**Comment by mathisscott** [comment]
@whizkidwwe1217 

For Clarity, we recommend the Z eye-tracking pattern for forms in confined blocks of space (modals, cards, and the wizard) and the F-pattern for the content area. Because the content area can be thin or very wide, right-aligned buttons in the content area are something we advise against.

Likewise with right-aligned labels. Left-aligned labels (even at the top of field) support the F and Z pattern better than right-aligned tables - which present a jagged left edge to eye-scanning.

I would want to know more about the use case of a design that mixed and matched left-aligned and right-aligned form labels. I don't understand why an app would want to do that.

As for vertical labels, our form fields go vertical on small device screens. There is perhaps a use case there where we could force that to happen at the form level despite screen size.

Our form fields work within grid columns can be used to achieve some layout variations: https://plnkr.co/edit/21PVKm?p=preview . That's the recommended method for getting the right edges of the form-fields to line up as in your second graphic.

We have an existing issue open for improving the way forms are laid out that might be of interest: #932 

In summary:
• Right-aligned buttons and labels are not patterns we support
• Pending UX's input, toggling left-aligned and vertically-aligned form field labels is something to consider
• Improving the forms-that-use-grid functionality is something to investigate, especially if #932 continues moving forward.

Thanks for the feedback, @whizkidwwe1217 .

**Comment by whizkidwwe1217** [comment]
@mathisscott  Z and F patterns will work for us. I would love to see those improvements in the future releases. Thanks for the feedback. You guys did a great job for making this awesome library. :-)
